### PR TITLE
Make sure consumer store is memory based when selected.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/consumer.go [hunk 1]
```diff
@@ -694,7 +694,12 @@ func (mset *stream) addConsumerWithAssignment(config *ConsumerConfig, oname stri
 	o.selectStartingSeqNo()
 
 	if !config.Direct {
-		store, err := mset.store.ConsumerStore(o.name, config)
+		streamStore := mset.store
+		// We now allow overrides from a stream being a filestore type and forcing a consumer to be memory store.
+		if config.MemoryStorage && streamStore.Type() == FileStorage {
+			streamStore = emptyMemStore
```
Comment (kozlovic) [inline server/consumer.go:700]: This seem to be a "shared" mem store that all consumers from any stream will share. Is that right? Is there code that may go from consumer's store back to that one and this is going to cause a mess? Meaning emptyMemStore.ConsumerStore() will return an instance of consumerMemStore but that one will reference the "global" emptyMemStore instance.. All consumerMemStore APIs (UpdateDelivered, UpdateAcks, etc..) will work on the same stream's memory store...

Comment (derekcollison) [reply server/consumer.go:700]: Only thing I saw was the consumer count being added and deleted, but they would all be shared there.

Comment (kozlovic) [reply server/consumer.go:700]: Yes, that's what I find out after looking more into it. So I guess for now this is safe. Alternatively, streams that are file-based could hold a new field that would be an empty memory store (created only when the file-based stream is created). Then here we would assign mset.memStore or something like that.

**Review by kozlovic — APPROVED** [review]
LGTM

**Review by kozlovic — APPROVED** [review]
LGTM

**Pull request merged**